editor/modules/runner/game-engine.ts

```diff
@@ -0,0 +1,143 @@
+        this.surface.setMouseHandler(this.input);
+        this.surface.addViewportListener(this.camera);
+    }
+
+    cleanUp() {
+        this.surface.removeViewportListener(this.camera);
+        this.surface.setKeyHandler();
+        this.surface.setMouseHandler();
+    }
+
+    play(map: Map) {
+        let chipsets_pos: {[path: string]: number} = {};
+        let chipsets_path: string[] = [];
+        map.fillChipsetsInfo(chipsets_pos, chipsets_path);
+
+        let map_tiled = this.surface.createTilesRenderEl();
+        map_tiled.loadTileLayerObject(chipsets_path, (chipsets, builder) => {
+            let handle = builder.setWidth(map.width)
+                .setHeight(map.height)
+                .tileSize(map.tile_size);
+            for (let i = 0; i < map.layers.length; ++i) {
+                let layer = map.layers[i].raw.map(pl => {
+                    return {
+                        tiles_id: pl.tiles_id,
+                        chipset: chipsets[chipsets_pos[pl.chipset]]
+                    };
+                });
+                handle.addLayer(layer);
+            }
+            // this.camera.centerOn(handle.build());
+        });
+        let player = this.surface.createSpriteRenderEl();
+        player.loadSpriteObject([125, 125, 125, 255], builder => {
+            this.player_handle = builder.buildWithSize(16, 24);
+            this.camera.centerOn(this.player_handle);
+        });
+        let hud = new Hud(this.player, this.lycan, this.surface);
+        let entities = this.surface.createSpriteBatchRenderEl();
+
+        let pipeline = new CommandBuffer([
+            DefaultFBO,
+            ClearAll,
+            new TileProgram(),
+            this.camera,
+            map_tiled,
+            FlipY,
+            new SpriteProgram(),
+            this.camera,
+            player,
+            new SpriteBatchProgram(),
+            this.camera,
+            entities,
+            hud,
+            () => {
+                let new_time = Date.now();
+                if (!this.last_time) {
+                    this.last_time = new_time;
+                }
+                let elapsed = (new_time - this.last_time) / 1000;
+                this.input.update(elapsed);
+                this.physics.update(elapsed);
+                // TODO: Move that elsewhere.
+                for (let i = 0; i < this.world.entities.length; ++i) {
+                    let {x, y} = this.world.entities[i].pos;
+                    this.sprites.setPos(i, x, y);
+                }
+                this.sprites.updatePositions();
+                this.player_handle.position(
+                    [this.player.pos.x, this.player.pos.y]
+                );
+                this.camera.centerOn(this.player_handle);
+                this.last_time = new_time;
+            }
+        ]);
+
+        this.surface.setActivePipeline(pipeline);
+        this.surface.focus();
+
+        // TODO(Nemikolh): Move this elsewhere.
```

Comment:
I think it's fine as is. I'm not sure exactly what I want to do about this code and I feel like creating an issue for it might be overkill.